src/test/java/uk/ac/ebi/embl/flatfile/reader/FeatureLocationsMatcherTest.java

```diff
@@ -77,31 +74,122 @@ public void testLocation() {
 
     location = "complement(<2807037..2807081)";
     assertEquals(location, test(location));
-    assertFalse(leftPartial(location));
-    assertTrue(rightPartial(location));
+    assertFalse(fivePrimePartial(location));
+    assertTrue(threePrimePartial(location));
 
     location = "complement(join(2182966..2183014,2183124..>2183128))";
     assertEquals(location, test(location));
-    assertFalse(leftPartial(location));
-    assertTrue(rightPartial(location));
+    // This is not a valid 5'
+    // assertTrue(fivePrimePartial(location));
+    assertFalse(threePrimePartial(location));
 
     location = "complement(join(<2182966..2183014,2183124..2183128))";
     assertEquals(location, test(location));
-    assertTrue(leftPartial(location));
-    assertFalse(rightPartial(location));
+    assertFalse(fivePrimePartial(location));
+    // This is not a valid 3'
+    // assertTrue(threePrimePartial(location));
+  }
+
+  @Test
+  public void testPartiality() {
+    /*
+        <2..3 → 5’ partial
+        2..>3 → 3’ partial
+        complement(<2..3) → 3’ partial
+        complement(2..>3) → 5’ partial
+        join(<2,4) → 5’ partial
+        join(2,>4) → 3’ partial
+        complement(join(<2,4)) → 3’ partial
+        complement(join(2,>4)) → 5’partial
+        join(complement(<2),4) → 3’ partial
+        join(2,complement(>4)) → 5’ partial
+        join(complement(<2),complement(4)) → 3’ partial
+        join(complement(2),complement(>4)) → 5’ partial
+    */
+
+    String location = "<2..3"; //  → 5’ partial
+    assertEquals(location, test(location));
+    assertTrue(fivePrimePartial(location));
+    assertFalse(threePrimePartial(location));
+
+    location = "2..>3"; //  → 3’ partial
+    assertEquals(location, test(location));
+    assertFalse(fivePrimePartial(location));
+    assertTrue(threePrimePartial(location));
+
+    location = "complement(<2..3)"; //  → 3’ partial
+    assertEquals(location, test(location));
+    assertFalse(fivePrimePartial(location));
+    assertTrue(threePrimePartial(location));
+
+    location = "complement(2..>3)"; //  → 5’ partial
+    assertEquals(location, test(location));
+    assertTrue(fivePrimePartial(location));
+    assertFalse(threePrimePartial(location));
+
+    location = "join(<2,4)"; //  → 5’ partial
+    assertEquals(location, test(location));
+    assertTrue(fivePrimePartial(location));
+    assertFalse(threePrimePartial(location));
+
+    location = "join(2,>4)"; //  → 3’ partial
+    assertEquals(location, test(location));
+    assertFalse(fivePrimePartial(location));
+    assertTrue(threePrimePartial(location));
+
+    location = "complement(join(<2,4))"; //  → 3’ partial
+    assertEquals(location, test(location));
+    assertFalse(fivePrimePartial(location));
+    // This is not a valid 3'
+    // assertTrue(threePrimePartial(location));
+
+    location = "complement(join(2,>4))"; //  → 5’partial
+    assertEquals(location, test(location));
+    // This is not a valid 5'
```

Comment:
- Please trust the location partialities in https://www.ebi.ac.uk/panda/jira/browse/ENA-6078
- This test is not correct at the moment.